### PR TITLE
Attest cleanup

### PR DESCRIPTION
**Pull request by drwells**
Lets see what the CI does!

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by drwells** [comment]
Hm, actually running the test suite has surprisingly poor performance. Let me investigate further before we merge.

**Comment by vibraphone** [comment]
@drwells It looks like you may need to kill an action that is still running? I don't think old actions are killed when new ones are created...

**Comment by drwells** [comment]
I think you're right. Let me manually kill any old ones.

**Comment by drwells** [comment]
Maintaining my own test runner is expensive but it's going to be necessary to keep up while we support autotools. I added a verbose mode - if you look at some of the old runs the majority of tests fail and I don't see why some would pass and others would fail. My best guess is that we are underlinked.

**Comment by drwells** [comment]
We get a lot of illegal instruction errors - that's our answer for `-march=native`.

**Comment by drwells** [comment]
@vibraphone we need to enable tirpc in the docker images so that we get XDR support. It's not documented but our libMesh tests rely on that for restart data.

**Comment by drwells** [comment]
We have tests running but the output for a lot of them is completely wrong. For example:
```
          IBFE/interpolate_velocity_01_2d.nodal_quadrature.b.mpirun=4.input  DIFF FAILED
  test failed with output:
      ----------------
      ##9       #:4   <== 8.7554918579035856396e-05
      ##9       #:4   ==> 1.6131799264655186654
      @ Absolute error = 1.6130923715e+0, Relative error = 1.8423777873e+4
      ##9       #:5   <== 0.00021638268562762519309
      ##9       #:5   ==> 3.0727148287267458393
      @ Absolute error = 3.0724984460e+0, Relative error = 1.4199372917e+4
```
The errors are really high in a way that doesn't make sense.

**Comment by drwells** [comment]
Furthermore, some of these timeouts don't make sense:
```
          IBFE/explicit_ex4_2d.scratch_hier.levels=1.mpirun=4.input ------ TIMEOUT
```
this test runs in 1.5 seconds on my 4-core desktop computer - it doesn't seem right that the virtual machine is 40x slower than my 8 year old workstation.

**Comment by vibraphone** [comment]
> @vibraphone we need to enable tirpc in the docker images so that we get XDR support. It's not documented but our libMesh tests rely on that for restart data.

@drwells OK. Do I need to pass libmesh's `./configure ...` line additional arguments or just make sure `tirpc-devel` is installed?

**Comment by drwells** [comment]
I think installing the package should be enough - libMesh doesn't have an extra flag to specify an install location.

**Comment by vibraphone** [comment]
> We get a lot of illegal instruction errors - that's our answer for `-march=native`.

Yes, the architecture of my linux box (that builds the 3rd party packages) is not necessarily the same as the runner that builds IBAMR. I can change the flags on the docker image (since I'm rebuilding libmesh and adding tirpc); do you have any suggestions on the flags to use? On our dashboard machines, we use `-march=core2 -mno-avx512f` if that helps.

**Comment by drwells** [comment]
I don't see any `-march=native` flags in the dockerfile - I don't think we need to change any compiler flags there.

There is something weird going on with libMesh that I do not understand - no parallel libMesh test passed. I'm not sure why.

**Comment by vibraphone** [comment]
> We have tests running but the output for a lot of them is completely wrong. For example ...

Have you been able to replicate by building in a local container (i.e., running `docker run -ti dcthomp/ibamr:0.9-build-deps bash` and building yourself)? If it helps, once the container is running, you can run the build script to build IBAMR.

**Comment by drwells** [comment]
Not yet - that's a good idea.

**Comment by vibraphone** [comment]
> I don't see any `-march=native` flags in the dockerfile - I don't think we need to change any compiler flags there.

@drwells The absence of `-march=native` in the Dockerfile doesn't mean the presumed architecture matches the github runners. We ran into that with our packaging process for modelbuilder.

> There is something weird going on with libMesh that I do not understand - no parallel libMesh test passed. I'm not sure why.

Well, as I just re-ran libmesh's configure step I see that it built without MPI. That is probably why.

**Comment by drwells** [comment]
That makes a lot of sense, actually - the partitioning tests all output the global range on every processor.

I'll put up some more libMesh checks in our `CMakeLists.txt` to try and catch these errors faster.

**Comment by vibraphone** [comment]
It's odd, because the Dockerfile passes mpicc/mpicxx to libmesh as the compilers to use.

**Comment by drwells** [comment]
libMesh's configuration script is somewhat buggy. For example, `--enable-petsc-required` isn't guaranteed to work - based on how you install PETSc it may not find all the files it needs and not end up including PETSc even with that flag.

**Comment by drwells** [comment]
Yet more reasons to switch to CMake, but you already probably agree with that sentiment :)

**Comment by vibraphone** [comment]
CMake is nice, but it doesn't prevent bugs.

**Comment by drwells** [comment]
Of course, but it's a lot harder to write bugs in CMake with targets et al than it is with m4 soup.

**Comment by drwells** [comment]
This branch got away from its original purpose to implement some necessary fixes (add a missing configuration option, implement verbose mode, do some general tidying up) of `attest`. @boyceg could you give this a quick look and approve? I'll move the CI script changes into a new PR.

File: attest [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-## This file is part of IBAMR.
-##
-## IBAMR is free software and is distributed under the 3-clause BSD
-## license. The full text of the license can be found in the file
-## COPYRIGHT at the top level directory of IBAMR.
-##
-## ---------------------------------------------------------------------
+# ---------------------------------------------------------------------
+#
+# Copyright (c) 2019 - 2020 by the IBAMR developers
```
Comment (boyceg) [inline attest:4]: Do we still need to update to 2021?

**Review by boyceg — APPROVED** [review]

File: attest (hunk 1 above)
Comment (drwells) [reply attest:4]: the update copyright release step will take care of that.

**Pull request merged**